src/blueprints/tv2/factories/tv2-graphics-action-factory.ts

```diff
@@ -134,5 +148,162 @@ export class Tv2GraphicsActionFactory {
+    switch (blueprintConfiguration.studio.GraphicsType) {
+      case Tv2GraphicsType.HTML: return this.createCasparCgFullscreen(blueprintConfiguration, actionManifests)
+      case Tv2GraphicsType.VIZ:
+      default: return this.createVizFullscreen(blueprintConfiguration, actionManifests)
+    }
+  }
+
+  private createVizFullscreen(blueprintConfiguration: Tv2BlueprintConfiguration, actionManifests: Tv2GraphicActionManifest[]): PartAction[] {
+    return actionManifests.map((manifest) => this.createVizGraphicsActionFromManifest(
+      blueprintConfiguration,
+      manifest
+    ))
+  }
+
+  private createVizGraphicsActionFromManifest(blueprintConfiguration: Tv2BlueprintConfiguration, manifest: Tv2GraphicActionManifest): PartAction {
+    const target: Tv2GraphicsTarget = Tv2GraphicsTarget.FULL
+    const fullGraphicPiece: PieceInterface = this.createVizFullGraphicsPiece(blueprintConfiguration, manifest)
+    const partInterface: PartInterface = this.createGraphicsPartInterface({
+      id: `${this.getGraphicTargetAsCamelString(target)}Part`, // Todo: make id unique
+      name: `${this.getGraphicTargetAsCamelString(target)} ${manifest.userData.name}`,
+      inTransition: {
+        keepPreviousPartAliveDuration: blueprintConfiguration.studio.VizPilotGraphics.KeepAliveDuration,
+        delayPiecesDuration: 0
+      },
+    })
+    return {
+      id: '',
+      name: '',
+      type: PartActionType.INSERT_PART_AS_NEXT,
+      data: {
+        partInterface: partInterface,
+        pieceInterfaces: [ fullGraphicPiece ]
+      }
+    }
+  }
+
+  private createVizFullGraphicsPiece(blueprintConfiguration: Tv2BlueprintConfiguration, manifest: Tv2GraphicActionManifest): PieceInterface {
+    return this.createGraphicsPieceInterface({
+      id: 'fullGraphicPiece', // Todo: make id unique
+      name: manifest.userData.name,
+      preRollDuration: blueprintConfiguration.studio.VizPilotGraphics.PrerollDuration,
+      pieceLifespan: this.findInfiniteModeFromConfig(blueprintConfiguration, manifest),
+      layer: Tv2SourceLayer.PILOT_GRAPHIC,
+      content: this.createVizFullGraphicsPieceContent(blueprintConfiguration, manifest)
+    })
+  }
+
+  private createGraphicsPartInterface(partInterfaceWithRequiredValues: Pick<PartInterface, 'id' | 'name'> & Partial<PartInterface>): PartInterface {
+    return {
+      disableNextInTransition: false,
+      expectedDuration: 0,
+      inTransition: {
+        keepPreviousPartAliveDuration: 0,
+        delayPiecesDuration: 0
+      },
+      isNext: true,
+      isOnAir: false,
+      isPlanned: false,
+      outTransition: {
+        keepAliveDuration: 0
+      },
+      pieces: [],
+      rank: 0,
+      segmentId: '',
+      ...partInterfaceWithRequiredValues
+    }
+  }
+
+  private getGraphicTargetAsCamelString(target: Tv2GraphicsTarget): string {
+    return target.toString().charAt(0) + target.toString().substring(1).toLowerCase()
+  }
```

Comment:
You are using this method twice. At the same place and even with the same value.
Instead of trying to making weird helper functions for stuff you don't know if you even need helper functions for, just take the simple approach.

This is what you use it for:
```
const target: Tv2GraphicsTarget = Tv2GraphicsTarget.FULL
...
const partInterface: PartInterface = this.createGraphicsPartInterface({
  id: `${this.getGraphicTargetAsCamelString(target)}Part`,
   name: `${this.getGraphicTargetAsCamelString(target)} ${manifest.userData.name}`,
   ...
```
`Tv2GraphicsTarget.FULL` evaluates to `FULL`, so you could just have written:
```
const target: string = Tv2GraphicsTarget.FULL.toString().toLowerCase()
...
const partInterface: PartInterface = this.createGraphicsPartInterface({
  id: `${target}Part`,
   name: `${target} ${manifest.userData.name}`,
   ...
```
and saved yourself a weird helper function.
(I would say that even the suggestion is overkill for what is actually happening here)

Reply:
Not to mention, your function doesn't even make it camelCase. It just lowercase everything except the first character...

Reply:
Method is gone.